pkg/cvo/availableupdates_test.go

```diff
@@ -0,0 +1,269 @@
+	return nil, nil
+}
+
+func (n notFoundConfigMapLister) Get(name string) (*corev1.ConfigMap, error) {
+	return nil, errors.NewNotFound(schema.GroupResource{Group: "", Resource: "configmap"}, name)
+}
+
+// osusWithSingleConditionalEdge helper returns:
+//  1. mock osus server that serves a simple conditional path between two versions.
+//  2. mock condition that always evaluates to match
+//  3. expected []ConditionalUpdate data after evaluation of the data served by mock osus server
+//     (assuming the mock condition (2) was used)
+//  4. current version of the cluster that would issue the request to the mock osus server
+func osusWithSingleConditionalEdge() (*httptest.Server, clusterconditions.Condition, []configv1.ConditionalUpdate, string) {
+	from := "4.5.5"
+	to := "4.5.6"
+	osus := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintf(w, `{
+  "nodes": [{"version": "%s", "payload": "payload/%s"}, {"version": "%s", "payload": "payload/%s"}],
+  "conditionalEdges": [
+    {
+      "edges": [{"from": "%s", "to": "%s"}],
+      "risks": [
+        {
+          "url": "https://example.com/%s",
+          "name": "FourFiveSix",
+          "message": "Four Five Five is just fine",
+          "matchingRules": [{"type": "PromQL", "promql": { "promql": "this is a query"}}]
+        }
+      ]
+    }
+  ]
+}
+`, from, from, to, to, from, to, to)
+	}))
+
+	updates := []configv1.ConditionalUpdate{
+		{
+			Release: configv1.Release{Version: to, Image: "payload/" + to},
+			Risks: []configv1.ConditionalUpdateRisk{
+				{
+					URL:     "https://example.com/" + to,
+					Name:    "FourFiveSix",
+					Message: "Four Five Five is just fine",
+					MatchingRules: []configv1.ClusterCondition{
+						{
+							Type:   "PromQL",
+							PromQL: &configv1.PromQLClusterCondition{PromQL: "this is a query"},
+						},
+					},
+				},
+			},
+			Conditions: []metav1.Condition{
+				{
+					Type:    "Recommended",
+					Status:  metav1.ConditionFalse,
+					Reason:  "FourFiveSix",
+					Message: "Four Five Five is just fine https://example.com/" + to,
+				},
+			},
+		},
+	}
+	mockPromql := &mock.Mock{
+		ValidQueue: []error{nil},
+		MatchQueue: []mock.MatchResult{{Match: true, Error: nil}},
+	}
+
+	return osus, mockPromql, updates, from
+}
+
+func newOperator(url, version string, promqlMock clusterconditions.Condition) (*availableUpdates, *Operator) {
+	currentRelease := configv1.Release{Version: version, Image: "payload/" + version}
+	registry := clusterconditions.NewConditionRegistry()
+	registry.Register("Always", &always.Always{})
+	registry.Register("PromQL", promqlMock)
+	operator := &Operator{
+		defaultUpstreamServer: url,
+		architecture:          "amd64",
+		proxyLister:           notFoundProxyLister{},
+		cmConfigManagedLister: notFoundConfigMapLister{},
+		conditionRegistry:     registry,
+		queue:                 queueStub{},
```

Comment:
You might be able to drop `queueStub` and use `workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())` like [this][1] here.  There's also some `clientProxyLister` and such in that file that I don't entirely understand, but which might be recyclable here to save some local-mocking effort.

[1]: https://github.com/openshift/cluster-version-operator/blob/53a5d57c2bd3c92c145dac005fab6e4cdcf8f43b/pkg/cvo/cvo_test.go#L2721

Reply:
:100: I got rid of the queue stub (not sure why I haven't tried to just instantiate a real queue...) but I kept the lister mocks, they are really simple.